packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -444,6 +467,30 @@ export abstract class SharedObjectCore<TEvent extends ISharedObjectEvents = ISha
      * when the op is ACKed or resubmitted, respectively
      */
     protected abstract applyStashedOp(content: any): unknown;
+
+    public emit(
+        event: EventEmitterEventType,
+        ...args: any[]): boolean {
+        let returnValue: boolean;
+
+        // Do not include the events emmited by SharedObject itself, we only care
+        // about the events from child classes.
+        if (["op", "pre-op"].includes(event.toString())) {
```

Comment:
This list would get out of date pretty soon as there is nothing enforcing keeping it up to date.
I'd suggest instead having function with different name (not "emit") for cases where we want to measure and report times.

Reply:
Makes sense. I did do that at some point but didn't like that I needed to change all DDS so they would receive an interface with the new function... I'll explore keeping `emit` as the one the child DDSs need to use, and maybe use a separate emitter that skips the measurements for `SharedObjectCore` itself.